.github/workflows/releases.yml

```diff
@@ -0,0 +1,103 @@
+name: Releases workflow
+
+on:
+  create:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+env:
+  GO_VERSION: '1.20.6'
+
+permissions:
+  contents: write
+  deployments: write
+
+jobs:
+  # TODO: Any way we can just call build.yml?
+  test:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - run: make
+      - run: make test
+      
+  integration:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        containerd: ["1.6.19", "1.7.0"]
+    env:
+      DOCKER_BUILD_ARGS: "CONTAINERD_VERSION=${{ matrix.containerd }}"
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.20.6'
+      - run: make integration
+
+  generate-artifacts:
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v4
+    - name: Setup and export variables
+      run: |
+        export release_tag=${GITHUB_REF#refs/*/} # Strip down to raw tag name
+        export release_version=${release_tag/v/} # Remove v from tag name
+        echo "release_version=$release_version" >> $GITHUB_ENV
+
+        echo "dynamic_binary_name=soci-snapshotter-${{ env.release_version }}-linux-amd64.tar.gz" >> $GITHUB_ENV
+        echo "static_binary_name=soci-snapshotter-${{ env.release_version }}-linux-amd64-static.tar.gz" >> $GITHUB_ENV
+        
+        mkdir release
+    - name: Create dynamic binary
+      run: |
+        make build
+        cp NOTICE.md THIRD_PARTY_LICENSES out/
+        cd out/
+        tar -czvf ../release/${{ env.dynamic_binary_name }} *
+        cd ../
+        rm -rf out/
```

Comment:
Is it worth moving some of this logic into a Make target or bash script for local runs? Thinking out loud, it might be helpful for troubleshooting. (Likewise for static build)

Reply:
I'm not against it but it's also only a few lines of pretty basic code that requires some basic preset conditions. I think this is good enough for now but if the process ever becomes more convoluted we should certainly revisit the idea.

That said, if we were to do that sooner rather than later, maybe the entire release logic should be put there.